src/host.rs

```diff
@@ -135,6 +345,30 @@ impl Host {
     pub(crate) fn tick(&mut self, now: Instant) {
         self.now = now;
 
+        if let Some(listener) = self.listener.as_ref() {
```

Comment:
Ah, more duplication! See previous comment.

Reply:
Yeah... embark, recv, release and tick are all pretty much copies. It should all be confined within host and we can restructure easily.